lib/types/src/types.rs

```diff
@@ -735,6 +738,12 @@ impl InstanceConfig {
         self.stack_limit = stack_limit;
         self
     }
+
+    /// Create instance configuration with profile.
+    pub fn with_profile(mut self) -> Self {
```

Comment:
The code responsible for instantiating a module could invoke a method to write out various data itself. I think making `write_perf_profiler_map` public would result in less convoluted code on both sides overall, compared to adding an `InstanceConfig` field.

Reply:
If go this way, we could make the next step and provide symbolic information instead, and make `write_perf_profiler_map` an utility function instead.

Reply:
I don't disagree, but I feel it may be better for that to be a future improvement. I have a suspicion that whatever other formats we might want to produce in the future, they would need more data, anyway.